### PR TITLE
HPCC-17739 Windows PTree crashes on valid XML due to unitialized values

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by afishbeck** [comment]
I haven't looked for other similar issues in the rest of the PTree code.

**Comment by afishbeck** [comment]
This is a regression in 6.4.

@richardkchapman please review

File: system/jlib/jptree.cpp [hunk 1]
```diff
@@ -2987,6 +2987,7 @@ void LocalPTree::setAttribute(const char *key, const char *val)
     {
         attrs = (AttrValue *)realloc(attrs, (numAttrs+1)*sizeof(AttrValue));
         v = &attrs[numAttrs++];
+        v->init();
```
Comment (afishbeck) [inline system/jlib/jptree.cpp]: Other option was to do a memset().

**Comment by richardkchapman** [comment]
If this is a regression in 6.4 then you need to target candidate-6.4.0 ...

The fix you are proposing suggests that the flag field in the union is not overlapping the pointer - and that is what needs fixing rather than initializing the flag separately.

**Comment by richardkchapman** [comment]
Do you have the ptree that illustrated the issue?

I can see a potential problem if LocalPTree::setAttribute was called with val==nullptr (but it's not specific to windows/32-bit if so)

**Comment by richardkchapman** [comment]
I see another bug too, but I can't see how this change would fix it.

**Comment by ghalliday** [comment]
@afishbeck is there a simple way to provoke the error?  richard has released a different fix, so I wanted to check it.

**Comment by richardkchapman** [comment]
Closing in favour of https://github.com/hpcc-systems/HPCC-Platform/pull/10071

**Pull request closed without merging**